quinn-proto/src/connection.rs

```diff
@@ -111,13 +111,18 @@ where
     first_1rtt_sent: Option<u64>,
     /// Sequence number of the first remote CID that we haven't been asked to retire
     first_unretired_cid: u64,
+    in_datagrams: VecDeque<Datagram>,
```

Comment:
How do you feel about adding all the datagram state into a separate `DatagramState` (to be bikeshedded) type that gets included in the connection? Presumably the borrowck situation would not be worse than it already is.

Reply:
Seems like a reasonable measure to keep `Connection`'s field count from growing too fast.